Fix U-boot RPMB regression (from 4.5.0 upstream)

ricardosalveti opened the pull request.

ricardosalveti commented on lib/libnxpse050/adaptors/apis/sss.c:
```diff
@@ -51,49 +54,123 @@ sss_policy_t se050_asym_policy = {
  * @return sss_status_t
  */
-sss_status_t se050_session_open(sss_se05x_ctx_t *ctx, bool encryption)
+#if defined(CFG_CORE_SE05X_SCP03_PROVISION) && CFG_CORE_SE05X_SCP03_PROVISION
```
Where is this config defined?

ldts replied:
would only be defined during build; Then it can be defined as 0 or 1 (I think is more user friendly this way).
currently I am using

 EXTRA_OEMAKE_append_imx8mmevk = " \
-    CFG_CORE_DYN_SHM=n CFG_DT=y CFG_OVERLAY_ADDR=0x43600000 \
+    CFG_CORE_DYN_SHM=n CFG_DT=y CFG_OVERLAY_ADDR=0x43600000 CFG_WERR=n CFG_CAAM_DBG=0x11\
+    CFG_CORE_SE05X=y CFG_NUM_THREADS=1 CFG_IMX_I2C=y\
+    CFG_CORE_SE05X_SCP03_PROVISION=0\
+    CFG_CORE_SE05X_DISPLAY_INFO=1\
+    CFG_CORE_SE05X_INIT_NVM=1\
+    CFG_CORE_SE05X_OEFID=0xA1F4 \
 "

ldts replied:
that would boot clearing the SE050 NVM - probably not what the normal user wants- not doing provisioning of new SCP03 keys and using the default SCP03 keys for a SE050 ARD board (OEFID is 0xA1F4)

etienne-lms committed:
[FIO fromtree] core: rpmb: handle not-implemented error code on device probing

Consider that non-secure world may report TEE_ERROR_NOT_IMPLEMENTED
error code on OPTEE_RPC_CMD_RPMB_PROBE_RESET RPC command when that
RPC command is not supported. This is needed to support U-Boot
that provides this return code, at least up to its release tag
v2025.01 [1].

Without this change, OP-TEE fails to communicate with U-Boot RPMB
RPC service since RPMB probe capability was merged in OP-TEE
(Fixes: tag below).

Link: https://source.denx.de/u-boot/u-boot/-/blame/v2025.01/drivers/tee/optee/supplicant.c?ref_type=tags#L96 [1]
Reported-by: Gavin Liu <gavin.liu@mediatek.com>
Closes: https://github.com/OP-TEE/optee_os/issues/7200
Fixes: 8dfdf3927214 ("core: rpmb: probe for kernel RPMB driver")
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>
(cherry picked from commit 96e8f740b7601089fcabb8806a06bbd704d33f3f)
Signed-off-by: Ricardo Salveti <ricardo@foundries.io>

etienne-lms committed:
[FIO fromtree] core: rpmb: fix mutex in directory populate

Fix mutex unlocking in rpmb_fs_dir_populate() that should protect
fat_entry_dir_deinit() execution.

Fixes: 5f68d7848fe8 ("core: RPMB FS: Caching for FAT FS entries")
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>
Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>
(cherry picked from commit b565152e9ce8c5cbfc0763fda10b0ddab5198d56)
Signed-off-by: Ricardo Salveti <ricardo@foundries.io>
